docs/source/examples.rst

```diff
@@ -0,0 +1,128 @@
+========
+Examples
+========
+
+The following examples are based on testing a SLES image using
+the test suite provided from the Open Build Service. For more information
+on installing the test suite see the :doc:`tests` documentation.
+
+Launch & Test a new instance
+============================
+
+To launch and test a new instance of an image provide the `--image-id` or
+`-i` option.
+
+The image ID will look different for all cloud providers. See the examples
+below for the three supported clouds:
+
+- Azure: SUSE:SLES:12-SP3:latest
+- EC2:   ami-0f7c9a39e20a9adea
+- GCE:   sles-12-sp3-v20180814
+
+To see what tests modules are available there is an `ipa list` command. You
+can invoke the command with `-v` option to see a verbose list of all tests
+within each module.
+
+.. code-block:: console
+
+   $ ipa list
+   test_sles_guestregister
+   test_sles_haveged
+   test_sles_hostname
+   test_sles_install_migration
+   test_sles_lscpu
+   test_sles_motd
+   test_sles_repos
+   test_sles
+
+Once you have a set of tests installed and chosen you can run ipa against an
+image. For this example we will test the Azure image and only run the base
+SLES tests:
+
+.. code-block:: console
+
+   $ ipa test azure -i SUSE:SLES:12-SP3:latest \
+     --region southcentralus \
+     --service-account-file /path/to/service_account.json \
+     --ssh-private-key-file /path/to/private_key_file \
+     --distro sles test_sles
+
+   Starting instance
+   Running tests /home/{user}/ipa/tests/test_sles.py
+   PASSED tests=1|pass=1|fail=0|error=0
+
+By default ipa will launch the instance with a small instance size. For Azure
+this is `Standard_B1ms`. Also, if the tests pass the instance will be
+terminated and resource group will be cleaned up in Azure. This behavior can
+be modified with the `--cleanup` and `--no-cleanup` options.
+
+There are many options available when running an ipa test which can be listed
+via the help command:
+
+.. code-block:: console
+
+   $ ipa test --help
+
+Test an existing instance
+=========================
+
+If you want to run tests on an existing instance you can provide the
+`--running-instance-id` or '-r'. All options and tests that are available
+for a new instance can be run against an existing one.
```

Comment:
Add the following based on the latest changes
"When testing a running instance the instance will not be removed when the tests pass."